src/main/java/net/sf/jabref/gui/exporter/SaveDatabaseAction.java

```diff
@@ -304,26 +315,58 @@ public void saveAs() throws Throwable {
             }
         }
 
-        File oldFile = panel.getBibDatabaseContext().getDatabaseFile().orElse(null);
-        panel.getBibDatabaseContext().setDatabaseFile(file);
+        BibDatabaseContext context = panel.getBibDatabaseContext();
+
+        if (context.getLocation() == DatabaseLocation.SHARED) {
```

Comment:
Now I moved the logic part as much as possible from here. But nevertheless there are multiple methods like [`run(...)`](https://github.com/JabRef/jabref/blob/master/src/main/java/net/sf/jabref/gui/exporter/SaveDatabaseAction.java#L114) or even worse like [`saveDatabase(...)`](https://github.com/JabRef/jabref/blob/master/src/main/java/net/sf/jabref/gui/exporter/SaveDatabaseAction.java#L166) which radically violate the _MVC pattern_. So this is a general problem in jabRef which requires a special attention.